### PR TITLE
Eran/MGMT-1413	Add basic identity functionality to bm-inventory

### PR DESCRIPTION
**Pull request by eranco74**

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -1325,6 +1325,8 @@ definitions:
       name:
         type: string
         description: Name of the OpenShift cluster.
+      owner:
```
Comment (ronniel1) [inline swagger.yaml:1328]: @avishayt Should this be in the swagger?
Maybe we also need the org?

**Comment by eranco74** [comment]
Subsystem:
SUCCESS! -- 45 Passed | 0 Failed | 0 Pending | 0 Skipped
System:
http://10.46.10.2:8080/job/test-infra/2029/

File: pkg/auth/auth.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package auth
+
+import (
+	"context"
+	"net/http"
+)
+
+type contextKey string
+
+const ContextUserIDKey = contextKey("user_id")
```
Comment (ronniel1) [inline pkg/auth/auth.go:10]: Context keys should usually be private to the module they are defined in
Also these values will all be in a struct in a single context key

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -1106,7 +1110,8 @@ func calculateHostNetworks(log logrus.FieldLogger, cluster *common.Cluster) []*m
 func (b *bareMetalInventory) ListClusters(ctx context.Context, params installer.ListClustersParams) middleware.Responder {
```
Comment (ronniel1) [inline internal/bminventory/inventory.go:1110]: Also need to update GetCluster, no?

Comment (eranco74) [reply internal/bminventory/inventory.go:1112]: Technically authorization should come from OCM.
In cluster list specifically we need to filter ourselves.
Also note that this is mainly a place holder and I can't really test it until we have auth.

Comment (ronniel1) [reply internal/bminventory/inventory.go:1112]: OCM will just tell us if the user has admin permissions or not.
We will need to filter based on that, in all relevant APIs

Comment (eranco74) [reply internal/bminventory/inventory.go:1112]: OK, I'll update, yet I'd like to start with just the list API and add more APIs in another PR

File: internal/identity/identity.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+func IsAdmin(ctx context.Context) bool {
+	return auth.UserRoleFromContext(ctx) == auth.AdminUserRole
+}
+
+func GetUserIDFilter(ctx context.Context) string {
+	query := ""
+	if !IsAdmin(ctx) {
+		username := auth.UserIDFromContext(ctx)
+		query = fmt.Sprintf("owner = '%s'", username)
```
Comment (tsorya) [inline internal/identity/identity.go:18]: no need for org? can't we have same user in different orgs?

File: internal/identity/identity.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+)
+
+func IsAdmin(ctx context.Context) bool {
+	return auth.UserRoleFromContext(ctx) == auth.AdminUserRole
+}
+
+func GetUserIDFilter(ctx context.Context) string {
+	query := ""
+	if !IsAdmin(ctx) {
+		username := auth.UserIDFromContext(ctx)
```
Comment (tsorya) [inline internal/identity/identity.go:17]: maybe userId ?

File: internal/identity/identity.go (hunk 4 above)
Comment (eranco74) [reply internal/identity/identity.go]: Unsure, I guess we need "org admin" and "super admin"

Comment (eranco74) [reply internal/identity/identity.go]: For now admin will get full access

**Pull request merged**